docs/registering-a-control-handler-function.md

```diff
@@ -90,3 +93,135 @@ int main(void)
     return 0;
 }
 ```
+
+## Listen with Hidden Window Example
+
+Per the remarks, if the gdi32.dll or user32.dll library are loaded,  **SetConsoleCtrlHandler** does not get called for the **CTRL\_LOGOFF\_EVENT** and **CTRL\_SHUTDOWN\_EVENT** events. The workaround specified is to create a hidden window by calling the [**CreateWindowEx**](/windows/win32/api/winuser/nf-winuser-createwindowexa) method with the *dwExStyle* parameter set to 0 and listen for the  [**WM\_QUERYENDSESSION**](/windows/win32/shutdown/wm-queryendsession) and [**WM\_ENDSESSION**](/windows/win32/shutdown/wm-endsession) window messages.
+
+It is also valid to have a visible window and listen for these messages. It is likely that in using gdi32.dll or user32.dll that the application would have a window. In that case, add the two messages to the existing [Window Procedure](/windows/win32/winmsg/using-window-procedures).
+
+More information on setting up a window and its messaging loop can be found at [Creating a Window](/windows/win32/learnwin32/creating-a-window).
+
+```C
+// CtrlHandler.cpp : This file contains the 'main' function. Program execution begins and ends there.
+//
+
+#include <windows.h>
+#include <stdio.h>
+
+BOOL WINAPI CtrlHandler(DWORD fdwCtrlType)
+{
+    switch (fdwCtrlType)
+    {
+        // Handle the CTRL-C signal.
+    case CTRL_C_EVENT:
+        printf("Ctrl-C event\n\n");
+        Beep(750, 300);
+        return TRUE;
+
+        // CTRL-CLOSE: confirm that the user wants to exit.
+    case CTRL_CLOSE_EVENT:
+        Beep(600, 200);
+        printf("Ctrl-Close event\n\n");
+        return TRUE;
+
+        // Pass other signals to the next handler.
+    case CTRL_BREAK_EVENT:
+        Beep(900, 200);
+        printf("Ctrl-Break event\n\n");
+        return FALSE;
+
+    case CTRL_LOGOFF_EVENT:
+        Beep(1000, 200);
+        printf("Ctrl-Logoff event\n\n");
+        return FALSE;
+
+    case CTRL_SHUTDOWN_EVENT:
+        Beep(750, 500);
+        printf("Ctrl-Shutdown event\n\n");
+        return FALSE;
+
+    default:
+        return FALSE;
+    }
+}
+
+LRESULT CALLBACK WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    switch (uMsg)
+    {
+    case WM_QUERYENDSESSION:
+    {
```

Comment:
You can remove these (and the ones below) if you'd like.

Reply:
The braces? Meh.